packages/ui/src/IconButton.tsx

```diff
@@ -1,183 +1,155 @@
+import {IconButtonProps} from "./Common";
+import {isMobileDevice} from "./MediaQuery";
 import {Modal} from "./Modal";
 import {Text} from "./Text";
+import {ThemeContext} from "./Theme";
 import {Tooltip} from "./Tooltip";
 import {Unifier} from "./Unifier";
+import {isNative} from "./Utilities";
 
-// eslint-disable-next-line react/display-name
-export const IconButton = forwardRef(
-  (
-    {
-      prefix,
-      icon,
-      onClick,
-      size,
-      type = "primary",
-      withConfirmation = false,
-      confirmationText = "Are you sure you want to continue?",
-      confirmationHeading = "Confirm",
-      tooltip,
-      indicator,
-      indicatorNumber,
-      indicatorStyle = {position: "bottomRight", color: "primary"},
-      testID,
-    }: IconButtonProps,
-    ref
-  ) => {
-    const [showConfirmation, setShowConfirmation] = useState(false);
+const ConfirmationModal: FC<{
+  visible: boolean;
+  title: string;
+  subTitle?: string;
+  text: string;
+  onConfirm: () => void;
+  onCancel: () => void;
+}> = ({visible, title, subTitle, text, onConfirm, onCancel}) => {
+  return (
+    <Modal
+      primaryButtonOnClick={onConfirm}
+      primaryButtonText="Confirm"
+      secondaryButtonOnClick={onCancel}
+      secondaryButtonText="Cancel"
+      subTitle={subTitle}
+      title={title}
+      visible={visible}
+      onDismiss={onCancel}
+    >
+      <Text>{text}</Text>
+    </Modal>
+  );
+};
 
-    // TODO: support hover states for icon button.
+const IconButtonComponent: FC<IconButtonProps> = ({
+  accessibilityLabel,
+  confirmationHeading = "Confirm",
+  confirmationText = "Are you sure you want to continue?",
+  iconName,
+  loading: propsLoading = false,
+  testID,
+  variant = "primary",
+  withConfirmation = false,
+  onClick,
+}) => {
+  const [loading, setLoading] = useState(propsLoading);
+  const [showConfirmation, setShowConfirmation] = useState(false);
+  const {theme} = useContext(ThemeContext);
 
-    const opacity = 1;
-    let bgColor: SurfaceColor;
-    let iconColor: IconColor;
-    // TODO match up icon colors better
-    if (type === "muted") {
-      bgColor = "base";
-      iconColor = "primary";
-    } else if (type === "secondary") {
-      bgColor = "neutralLight";
-      iconColor = "secondaryDark";
-    } else {
-      bgColor = "primary";
-      iconColor = "inverted";
-    }
+  if (!theme) {
+    return null;
+  }
 
-    const IndicatorPosition = {
-      bottomRight: {bottom: "20%", right: "20%"},
-      bottomLeft: {bottom: "20%", left: "20%"},
-      topRight: {top: "20%", right: "20%"},
-      topLeft: {top: "20%", left: "20%"},
-    };
+  const getStyles = () => {
+    let backgroundColor = theme.surface.primary;
+    let color = theme.text.inverted;
 
-    const indicatorPosition = {position: "absolute", ...IndicatorPosition[indicatorStyle.position]};
+    if (variant === "secondary") {
+      backgroundColor = theme.surface.neutralLight;
+      color = theme.surface.secondaryDark;
```

Comment:
This is a question out of curiosity but do we want to stick with theme.text for color in this case or were we not able to find the specific question within the text theme object?

Reply:
The values are defined as it is represented in Figma. I had pointed this out to josh and it is okayed. The button using surface as `color` is pretty standard when building libraries. The button should be a combination of natural colors (the apps background colors) and highlights.
By constructing in this way if surfaceDark is changed meaning darker or different color palette this will correctly reflect global changes as expected. So if we support dark mode opposite secondaryDark Surface and Opposite inverted should show 100% inverted
***
This logic is specific exclusive to this component and should not be shared. As each element has sorta it's own rules and functionality.
So getStyles can be present in many components if need be, but the properties being defined (i.e. the return) will not always be the same

Reply:
That makes sense!